src/execution.jl

```diff
@@ -80,65 +110,56 @@ session-dependent objects (e.g., a `CuModule`).
 function cached_compilation(cache::AbstractDict{<:Any,V},
```

Comment:
This would be the API breakage. We need to change this to `AbstractDict{<:Any, CacheEntry}{V}}` or take `V` as a separate parameter.